app/src/main/java/com/guzelgimadieva/tasky/authorization/ui/register/RegisterScreenViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.guzelgimadieva.tasky.authorization.ui.register
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterScreenViewModel @Inject constructor() : ViewModel() {
+    private val _registerState = MutableStateFlow(RegisterScreenState())
+    val registerState: StateFlow<RegisterScreenState> = _registerState.asStateFlow()
+
+    fun onEvent(event: RegisterEvent) {
+        when (event) {
+            is RegisterEvent.UsernameChanged -> {
+                _registerState.update { _registerState.value.copy(username = event.username) }
+            }
+            is RegisterEvent.EmailChanged -> {
+                _registerState.update { _registerState.value.copy(email = event.email) }
+            }
+            is RegisterEvent.PasswordChanged -> {
+                _registerState.update { _registerState.value.copy(password = event.password) }
```

Comment:
When using update, it's intended that you use the state instance the update block gives you rather than referring to the global value. This is what I mean:
```
_registerState.update {
    it.copy(...) // Instead of registerState.value.copy(...)
}
```